polyseam/cndi
# [Template]: Minio

# Related issue
#401 




[#33333](https://github.com/polyseam/cndi/issues/33333)

# Description

Minio template based on the minio operator https://github.com/minio/operator/tree/master

# Test Instructions
Use this script to test connection. 
Make sure to update the url, username and password.
```
import os
from minio import Minio
from minio.error import S3Error
from dotenv import load_dotenv

def test_minio_connection(endpoint, access_key, secret_key):
    client = Minio(
        endpoint,
        access_key=access_key,
        secret_key=secret_key,
        secure=True  # Set this to True if your MinIO server uses HTTPS
    )

    # Attempt to list buckets to test the connection
    try:
        buckets = client.list_buckets()
        print("Connection Successful. Buckets List:")
        # for bucket in buckets:
        print(buckets)
    except S3Error as err:
        print("Connection Error:", err)
    # client.fput_object(bucket_name='my-crd-bucket',file_path='./README.md', object_name="cndi")
    #client.fget_object(bucket_name='my-crd-bucket',file_path='~/Documents/README.md',object_name="cndi",)

if __name__ == "__main__":
    endpoint = 'tenant.url.com'  # Example: "localhost:9000"
    access_key = 'username'
    secret_key = 'password'

    test_minio_connection(endpoint, access_key, secret_key)
```



# Code of Conduct

By submitting this Pull Request, you agree to follow our
[Code of Conduct](https://github.com/polyseam/cndi/blob/main/CODE_OF_CONDUCT.md)


# Notes (Optional)
